bcgov/cas-reporting
# Chore/68 Initialize Backend

Addresses #68 - [Project link](https://github.com/orgs/bcgov/projects/123/views/1?pane=issue&itemId=49861452)
Adds Terraform scripts and Helm charts to run them. Resource states migrated from Terraform Cloud to a Google Cloud Storage bucket.

## Changes 🚧

- Added Python, Black (linting), Pytest, Poetry, and configured Poetry
- Added Django and Django-Ninja
  - Initialized Django app for Reporting
  - Set up test endpoint
  - Set up basic test for Pytest
- Added example .env file
- Added Makefile and developer environment setup instructions


## To test 🔬

- Go to `docs/developer-environment-setup.md` and follow the directions to set up your developer environment to set up the backend.
- Start up the local Postgres server and start Django (following directions from the documents). 
- Navigate to `localhost:8000/api/docs`. It should show the Ninja API documentation for _add_.
- Navigate to the `localhost:8000/api/add?a=4&b=2` endpoint in your browser. It should display the result of the addition.

## Notes📝 

- All of the setups has been done to be consistent with `cas-registration`. Changes that might have been made later in the backend should be reflected here where it makes sense to get them in at the start. 
- We've been discussing having a single Django project instance across all of BCIERS, but still split into different apps. Will we want to package parts of this backend once we start developing it and add it as a submodule/package in [cas-registration](https://github.com/bcgov/cas-registration)?
